sql-odbc/src/sqlodbc/opensearch_communication.cpp

```diff
@@ -1005,7 +1009,9 @@ std::string OpenSearchCommunication::GetClusterName() {
 
 void OpenSearchCommunication::SetSqlEndpoint() {
     std::string distribution = GetServerDistribution();
-    if (distribution.compare("opensearch") == 0) {
+    if (distribution.empty()) {
+        sql_endpoint = "Error";
```

Comment:
If this is an error state, is there a better way to propagate the error to the user?

Is it possible for `distribution` to be `empty`?

Reply:
`Distribution` is empty in the event of `GetServerDistribution()` having an exception caught.

Reply:
I see.

So, with change, if `GetServerDistribution` return `""`, `SetSqlEndpoint` sets `sql_endpoint` to `"Error"`, then `EstablishConnection` proceeds to try to connect to `Error`, probably fails and returns false?